nilearn/tests/test_signal.py

```diff
@@ -195,8 +240,13 @@ def test_butterworth(rng):
         )
     assert not np.array_equal(data, out)
 
-    # Test check for frequency lower than allowed (<0).
-    # The frequency should be modified and the filter should be run.
+
+def test_butterworth_warnings_HPF_too_low(data_butterworth_single_timeseries):
```

Comment:
that's the only test that's now failing with this change: need to figure out why

Reply:
the negative high_pass looks spurious.

Reply:
I thought so as well but actually it's there to test this (set high pass filter to a minimal value)

https://github.com/nilearn/nilearn/blob/62d8d5c3861ffb8fcd2b2e513958d40b29797032/nilearn/signal.py#L312-L318

The thing I am not clear about in terms of "design" is why reset a negative value to some very small value instead of just setting it to 0?

Reply:
git blamed back to PRs:
- https://github.com/nilearn/nilearn/pull/1656
- https://github.com/nilearn/nilearn/pull/1629

Reply:
I guess that 0 would lead to crazy outcomes for some filters (at least in previous scipy implementations...).